content/advent-2015/vendor-folder.md

```diff
@@ -0,0 +1,167 @@
+was that each of these were different. The individual problems included;
+
+ * Many people did not like modifying the import paths or being required to
+   include dependent packages in their repository.
+ * Modifying `GOPATH` implied using a bare `go build` would not be sufficient.
+   Wrappers for the `go` command emerged, each slightly different.
+ * Modifying packages in the normal `GOPATH` required each project
+   to have a unique `GOPATH`.
+
+## A solution to these problems
+
+With Go 1.5 a new method to discover go packages was released.
+Nothing has been changed or added
+to the go language or the go compilers. Packages must still reside in `GOPATH`.
+However, if a package or a parent folder of a package contains folder named
+`vendor` it will be searched for dependencies using the `vendor` folder as an
+import path root. While `vendor` folders can be nested, in most cases it is
+not advised or needed.
+Any package in the `vendor` folder will be found *before* the standard library.
+
+To enable this in Go 1.5 set the environment variable `GO15VENDOREXPERIMENT=1`.
+In Go 1.6 this will be on by default without an environment variable.
+
+## An example
+
+This simple package lives in `$GOPATH/src/github.com/kardianos/spider`.
+```
+$ cat main.go
+package main
+
+import (
+	"bytes"
+	"flag"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/andybalholm/cascadia"
+	"github.com/tdewolff/parse/css"
+	"golang.org/x/net/html"
+)
+...
+```
+
+```
+$ tree
+.
+├── css_test.go
+├── main.go
+└── vendor
+    ├── github.com
+    │   ├── andybalholm
+    │   │   └── cascadia
+    │   │       ├── LICENSE
+    │   │       ├── parser.go
+    │   │       ├── README.md
+    │   │       └── selector.go
+    │   └── tdewolff
+    │       ├── buffer
+    │       │   ├── buffer.go
+    │       │   ├── lexer.go
+    │       │   ├── LICENSE.md
+    │       │   ├── reader.go
+    │       │   ├── README.md
+    │       │   ├── shifter.go
+    │       │   └── writer.go
+    │       └── parse
+    │           ├── common.go
+    │           ├── css
+    │           │   ├── hash.go
+    │           │   ├── lex.go
+    │           │   ├── parse.go
+    │           │   ├── README.md
+    │           │   └── util.go
+    │           ├── LICENSE.md
+    │           ├── README.md
+    │           └── util.go
+    ├── golang.org
+    │   └── x
+    │       └── net
+    │           └── html
+    │               ├── atom
+    │               │   ├── atom.go
+    │               │   ├── gen.go
+    │               │   └── table.go
+    │               ├── const.go
+    │               ├── doc.go
+    │               ├── doctype.go
+    │               ├── entity.go
+    │               ├── escape.go
+    │               ├── foreign.go
+    │               ├── node.go
+    │               ├── parse.go
+    │               ├── render.go
+    │               └── token.go
+    └── vendor.json
```

Comment:
This gives the impression that `vendor.json` is part of the `vendor/` folder feature in Go. Since this is part of govendor and vendor-spec it would clarify the link if you explicitly called it out. For someone not familiar with the situation it might be confusing.